core/src/main/java/com/linecorp/armeria/server/ArmeriaDefaultHttpRequest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Map;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.RequestHeadersBuilder;
+
+import io.netty.handler.codec.http.DefaultHttpMessage;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.util.internal.ObjectUtil;
+
+final class ArmeriaDefaultHttpRequest extends DefaultHttpMessage implements HttpRequest {
```

Comment:
Is there a specific reason to extend `DefaultHttpMessage`? That internally creates `DefaultHttpHeaders` again. Duplicate headers may result in performance degression. 
```suggestion
final class ArmeriaHttp1Request implements HttpRequest {
```